specs/agents/span-links.md

```diff
@@ -0,0 +1,36 @@
+## Span Links
+
+A Span or Transaction MAY be linked to zero or more other Spans/Transactions that are causally related.
+
+Specific use-cases for Span Links:
+
+1. When a single transaction represents the batch processing of several messages, the agent SHOULD be able to link back to the traces that have produced the messages.
+2.  When the agent receives a `traceparent` header from outside a trust boundary, it SHOULD restart the trace (creating a different trace id with its own sampling decision) and link to the originating trace.
+3. Close gap for the OTLP intake - [OTel's specification of span links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans)
+
+Spans and Transactions MUST collect links in the `links` array with the following fields on each item:
+- `trace.id`: the id of the linked trace.
+- `span.id`: the id of the linked span or transaction.
+
+Example:
+
+```
+ "links": [
+      {"trace": {"id": "traceId1"}, "span": {"id": "spanId1"}},
+      {"trace": {"id": "traceId2"}, "span": {"id": "spanId2"}},
+    ]
+```
+
+## Trace Continuation
+
+We can expect incoming requests into an application with our agent which contains a `traceparent` header added by a 3rd party component. In this situation we end up with traces where part of the trace is outside of our system.
+
+In order to handle this properly, the agent SHOULD implement a trace continuation strategy.
+
+The agent SHOULD offer a configuration called `trace_continuation_strategy` with the following values and behavior:
+
+- `continue_always`: The agent takes the `traceparent` header as it is and applies it to the new transaction.
+- `restart_always`: The agent always creates a new trace with a new trace id. In this case the agent MUST create a link in the new transaction pointing to the original trace.
+- `restart_external`: The agent first checks the `tracecontext` header. If the header contains the `es` vendor flag, it's treated as internal, otherwise (including the case when the `tracecontext` header is not present) it's treated as external. In case of external calls the agent MUST create a new trace with a new trace id and MUST create a link in the new transaction pointing to the original trace.
+
+The default is `continue_always`. 
```

Comment:
Is this true for the .NET agent? IIRC, it does something similar to `restart_external`.

Reply:
> Is this true for the .NET agent? 

Yes, it is. The default is still the original behaviour: just get he value and apply it.

> IIRC, it does something similar to restart_external.

Correct, but that's not the default - [there is the `TraceContextIgnoreSampledFalse` flag](https://www.elastic.co/guide/en/apm/agent/dotnet/current/config-supportability.html#config-trace-context-ignore-sampled-false), but by default it's turned off. So the suggested default here would be fine for .NET, that's the same as the current default.

As discussed `TraceContextIgnoreSampledFalse` will be deprecated long term after this spec is implemented, and then as you say, `restart_external` will be basically the same as the deprecated flag. (Not 100% same, because in .NET we still reused the trace id, we just removed the parent id and made a new sampling decision, but I don't think this changes anything...).